instrumentation/aws-sdk/aws-sdk-1.11/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/awssdk/v1_11/SqsParentContext.java

```diff
@@ -34,7 +34,7 @@ static Context ofSystemAttributes(Map<String, String> systemAttributes) {
     String traceHeader = systemAttributes.get(AWS_TRACE_SYSTEM_ATTRIBUTE);
     return AwsXrayPropagator.getInstance()
         .extract(
-            Context.current(),
+            Context.root(),
```

Comment:
I am trying to understand this change. Is there a test that failed before but now passes?

Reply:
It's the second point of the description :)

Basically with context.current() the model for consuming a message without parent tracing attribute would be:
- client (http receive) span parent of consumer (messaging) span
while with parent tracing attribute:
- producer (sending) span parent of all consumer (messaging) span
- client (http receive) span decoupled (modelling http receive operation)

With this change modelling is the same no matter if the attribute is present, that is client (http receive) span is always decoupled from consume (messaging) span.

@iNikem does this ^^ make sense for you now?

Reply:
Partially :D I still don't understand it fully, but now at least I have a sense, that you did the right thing :)

Reply:
@kubawach Is it possible to create the consumer span after the client span is closed?

Basically we can imagine this 

------- Server ---------------------
---Recieve--- ---Consume---

I think we need the consume span to link to server right?

Though I recall a very similar conversation with @trask and @inikem a few months ago and can't remember what we decided.

Reply:
@anuraaga yep, we could definitely change the order in TracingRequestHandler#afterResponse so that consume spans are simply created after client is closed. For this particular scenario would yield the same results (no bogus parent-child between client and consume in case of lacking AWS tracing parent attribute).

With regards to linking - currently consume is linked to server (produce) - provided the info is available as a system attribute of a message. Linking "client" with "server" (produce) however would be much harder to do, as "client" is created upon http request when the parent tracing info is not yet available. AFAIK our API does not allow to set the parent later on.

To sum up - if you prefer to stay with Context.current() for  SqsParentContext#ofSystemAttributes I can switch the code to create consumer spans after client is closed :)

Reply:
I may not be fully following, but if it helps, we use `Context.root()` in other messaging instrumentation (e.g. JMS, Kafka, RabbitMQ) for message listeners which have messages pushed to them, treating them as entry points (similar to http server entry points where we also use `Context.root()`)